docs/java/reference/messaging/MessageRepository.mdx

```diff
@@ -0,0 +1,115 @@
+---
+sidebar_position: 0
+---
+
+# MessageRepository
+
+The `MessageRepository` contains method used to create, edit, get and delete message using the Java Wrapper Library
+
+## Constructors
+
+### constructor
+
+- **new MessageRepository(String projectId, String apiToken, String spaceName)**
+
+**Parameters**
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| `projectId` | `String` | ProjectId gotten from your Signalwire space e.g `a10d8a9f-2166-4e82-56ff-118bc3a4840f` |
+| `apiToken` | `String` | Api token generated from your Signalwire dashboard e.g `PT9e5660c101cd140a1c93a0197640a369cf5f16975a0079c9`|
+| `spaceName` | `String` | Name of your space e.g `testing` |
+
+## Methods
+
+### create
+
+▸**create(**`String to`, `String from`, `String body`**)**:`NewMessageResponse`
+
+Method to create a message
+
+**Parameters**
+
+|Name|Type|Description|
+| --- | --- | ---|
+| `to` | `String` | The phone number in E.164 format that received the message |
+| `from` | `String` | The phone number in E.164 format. For inbound messages, this will be the remote phone number who sent the message. |
+| `body` | `String` | The text of the message. Up to 1600 characters long and can be null if no message was sent. |
+
+#### Returns
+`NewMessageResponse`
+
+### create(MessageRequest request)
+
+▸**create(**`MessageRequest request`**)**:`NewMessageResponse`
+
+Create a message with additional information
+
+**Usage**
+
+```java
+MessageRequest request = new MessageRequest.Builder()
+                        .to("123yyyxxxx")
+                        .from("13456yyyuuu")
+                        .body("Hello world")
+                        .mediaUrl("example.com/test.png")
+                        .build();
+
+// Send message
+NewMessageResponse response = repository.create(request);
+
+// Print SID
+System.out.println(response.sid);
+```
+
+#### Returns
+`NewMessageResponse`
+
+### getMessages
+
+▸**getMessages()**:`Messages`
+
+Get all the messages on your Signalwire project
+
+#### Returns
+`Messages`
+
+### getMessages(MessageFilter filter)
```

Comment:
to make the title shorter, could we change this (and any other similar case) to just display the type of the parameter? E.g.

    ### getMessages(MessageFilter)

just for the header (the line below is fine)